fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/AgilityContextTest.java

```diff
@@ -0,0 +1,217 @@
+                "But if you’re a worm, sleep late.";
+        try (FDBRecordContext context = openContext(insertProps)) {
+            final AgilityContext agilityContext = AgilityContext.factory(context, true);
+            for (int i = 0; i < loopCount; i++) {
+                byte[] key = Tuple.from(2023, i).pack();
+                byte[] val = Tuple.from(i, ShelSilverstein).pack();
+                agilityContext.set(key, val);
+            }
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext(insertProps)) {
+            final AgilityContext agilityContext = AgilityContext.factory(context, false);
+            for (int i = 0; i < loopCount; i++) {
+                byte[] key = Tuple.from(2023, i).pack();
+                byte[] val = Tuple.from(i, ShelSilverstein).pack();
+                agilityContext.get(key).thenApply(bytes -> {
+                    assertEquals(val, bytes);
+                    return null;
+                });
+            }
+        }
+    }
+
+    @Test
+    void testAgilityContextSizeLimit() {
+        testAgilityContextOneLongWrite(73, 100, 100000);
+    }
+
+    @Test
+    void testAgilityContextTimeLimit() {
+        testAgilityContextOneLongWrite(77, 100000, 10);
+    }
+
+    void napTime(int napTimeMilliseconds) {
+        try {
+            Thread.sleep(napTimeMilliseconds);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testAgilityContextAtomicAttribute(boolean useAgile) {
+        final long prefix = 0x1abc1356; // avoid other tests' data
+        IntStream.rangeClosed(0, threadCount).parallel().forEach(threadNum -> {
+            try (FDBRecordContext context = openContext()) {
+                final AgilityContext agilityContext = AgilityContext.factory(context, useAgile);
+                for (long i = 1700; i < 1900; i += 17) {
+                    final long iFinal = i;
+                    if (threadNum == 0) {
+                        agilityContext.accept(aContext -> {
+                            final Transaction tr = aContext.ensureActive();
+                            for (int j = 0; j < 5; j++) {
+                                tr.set(Tuple.from(prefix, iFinal, j).pack(),
+                                        Tuple.from(iFinal).pack());
+                            }
+                        });
+                        napTime(3); // give a chance to other threads to run
+                    } else {
+                        napTime(3);
+                        agilityContext.accept(aContext -> {
+                            long[] values = new long[5];
+                            final Transaction tr = aContext.ensureActive();
+                            for (int j = 4; j >= 0; j--) {
+                                byte[] val = tr.get(Tuple.from(prefix, iFinal, j).pack()).join();
```

Comment:
I don't think it's particularly relevant for lucene, but we may, at some point, want to change `accept` to return a `CompletableFuture<Void>`

Reply:
👍 
`accept` and `acceptAsync`?